### PR TITLE
addr

### LINKED ISSUE #15741: addr

### PR DESCRIPTION
**Pull request by a00ayad00**
Close #15741

### CODE REVIEW

File: ivy/functional/frontends/tensorflow/math.py [hunk 1]
```diff
@@ -673,3 +673,8 @@ def in_top_k(target, pred, k, name=None):
 @to_ivy_arrays_and_back
 def conj(x, name=None):
     return ivy.conj(x)
+
+
```
Comment (paulaehab) [inline ivy/functional/frontends/tensorflow/math.py:677]: please remove changes in this file

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 2]
```diff
@@ -2523,7 +2523,7 @@ def test_tensorflow_in_top_k(
         test_flags=test_flags,
         fn_tree=fn_tree,
         on_device=on_device,
-        targets=x[0],
+        target=x[0],
```
Comment (paulaehab) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:2526]: please remove changes in this file

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 3]
```diff
@@ -8427,5 +8427,63 @@ def test_torch_instance_is_complex(
...
+    )
+    
+    
+# addr
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="addr",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
```
Comment (paulaehab) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8441]: please add test for all valid types

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 4]
```diff
@@ -8427,5 +8427,63 @@ def test_torch_instance_is_complex(
...
+    frontend_method_data,
+    init_flags,
+    method_flags,
+    on_device,
+):
+    input_dtype, x, vec1, vec2 = dtype_and_matrices
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtype,
+        init_all_as_kwargs_np={
+            "data": x[0],
```
Comment (paulaehab) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:8473]: we don't write this part of test like that please take a look at similar tests in file

**Review by paulaehab — CHANGES_REQUESTED** [review]
After you resolve changes please make sure that all tests pass

**Comment by paulaehab** [comment]
> Hi, @paulaehab It gives me this error 😢😢
> 
> FAILED ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_addr[cpu-ivy.functional.backends.numpy-False-False] - RuntimeError: input must be 1D vectors
> 
> FAILED ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_addr[cpu-ivy.functional.backends.tensorflow-False-False] - RuntimeError: input must be 1D vectors
> 
> FAILED ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_addr[cpu-ivy.functional.backends.torch-False-False] - RuntimeError: input must be 1D vectors
> 
> 3 failed, 226 warnings in 254.05s (0:04:14)

@a00ayad00 Hi, could you take a look at this test case, I think it will help you in testing this function it will pass most of the frameworks, Also take care about the difference between the two test strategies

**Comment by a00ayad00** [comment]
I almost done @paulaehab ❤️
I really want to thank you for guiding me.
I just got that error bellow:

FAILED ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_addr[cpu-ivy.functional.backends.tensorflow-False-False] - ivy.utils.exceptions.IvyError: tensorflow: promote_types_of_torch_inputs: tensorflow: nested_map: Tensor conversion...
1 failed, 2 passed, 50 warnings in 184.23s (0:03:04)

**Comment by a00ayad00** [comment]
Yes @paulaehab 
I soled the problem already and I got TEST PASSED !

**Comment by paulaehab** [comment]
Woow it looks great for me @a00ayad00 😄 , there is a small change need to be done please remove  changes in math files and I will merge the PR for you as the PR must contain changes in files related to function only 😄

**Comment by paulaehab** [comment]
@a00ayad00 After removing changes in math files I will merge it 😄

**Comment by a00ayad00** [comment]
I already removed the changes, maybe !!
But any way, thanks a lot ya ehab ❤️❤️
I implement a function in that math file, but I have some errors too in it and I can't implement it a week and half ago 😢
I hope you can help me in that and I hope too to be my assignee in all the next PRs because there is no much who can help me like that like you 😅💖💖💖

**Comment by paulaehab** [comment]
> Thanks for this words 😄  but I need you to remove this changes from this branch as you can see there is change in math.py you add new line could you make a new commit remove it or save you function and revert the commits until there no files except the files related to the function

File: ivy/functional/frontends/tensorflow/math.py [hunk 5]
```diff
@@ -672,4 +672,4 @@ def in_top_k(target, pred, k, name=None):
 
 @to_ivy_arrays_and_back
 def conj(x, name=None):
-    return ivy.conj(x)
+    return ivy.conj(x)
```
Comment (paulaehab) [inline ivy/functional/frontends/tensorflow/math.py:675]: remove this change and these file from branch

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py [hunk 6]
```diff
@@ -2552,4 +2552,4 @@ def test_tensorflow_conj(
         fn_tree=fn_tree,
         on_device=on_device,
         x=x[0],
-    )
```
Comment (paulaehab) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:2555]: Also remove this change from the branch and the PR

**Review by paulaehab — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py (hunk 6 above)
Comment (a00ayad00) [reply ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py:2555]: ok

File: ivy/functional/frontends/tensorflow/math.py (hunk 5 above)
Comment (a00ayad00) [reply ivy/functional/frontends/tensorflow/math.py:675]: But how can I fix these simple changes
In this change I removed and put the line again without any change 😅
I did all these changes unintentionally

**Comment by a00ayad00** [comment]
I'll discard all these changes and reopen the PR again, just wait...

**Comment by paulaehab** [comment]
@a00ayad00  ask me for review when you open new PR

**Comment by a00ayad00** [comment]
Hey @paulaehab 
I did it 😅😅👌👌

**Comment by paulaehab** [comment]
@a00ayad00  Great Work 🙂, merged

**Comment by a00ayad00** [comment]
Hope see you again, ya ehap 💖💖💖

**Pull request merged**